cloudify_agent/installer/runners/winrm_runner.py

```diff
@@ -470,12 +483,50 @@ def close(self):
         pass
 
     def run_script(self, script_path):
-        remote_path = self.put_file(script_path)
-        result = self.run(remote_path)
+        remote_path = ntpath.join(
+            self.get_temp_dir(),
+            ntpath.basename(script_path),
+        )
+        self.put_file(script_path, remote_path)
+        result = self.run(remote_path, powershell=True)
         self.delete(ntpath.dirname(remote_path))
         return result
 
 
+def split_into_chunks(contents):
+    """Split content into chunks to avoid command line too long error.
+
+    Maximum allowed commmand line length should be 2047 in old windows:
+    https://support.microsoft.com/en-us/help/830473/command-prompt-cmd--exe-command-line-string-limitation
+
+    :param contents:
+        The contents of a file that exceeds the maximum command line length in
+        windows.
+    :type content: str
+    :returns: The same content in chunks that won't exceed the limit
+    :rtype: list[str]
+
+    """
+    # Split content into chunks to avoid command line too long error
+    # maximum allowed commmand line length should be 2047 in old windows:
+    # https://support.microsoft.com/en-us/help/830473/command-prompt-cmd--exe-command-line-string-limitation
+    max_size = 2000
+    separator = '\r\n'
+
+    def join_lines(lines, line):
```

Comment:
This is pretty neat! But I think I'd like a unittest for this particular function - just to see that we get exactly the results we want. And maybe play around with different strings and escape characters inside the split string